easybuild/framework/easyconfig/tools.py

```diff
@@ -465,3 +467,61 @@ def review_pr(pr, colored=True, branch='develop'):
             lines.extend(['', "(no related easyconfigs found for %s)\n" % os.path.basename(ec['spec'])])
 
     return '\n'.join(lines)
+
+
+def dump_env_script(easyconfigs):
+    """
+    Dump source scripts that set up build environment for specified easyconfigs."""
+    ecs_and_script_paths = []
+    for easyconfig in easyconfigs:
+        script_path = '%s.env' % os.path.splitext(os.path.basename(easyconfig['spec']))[0]
+        ecs_and_script_paths.append((easyconfig['ec'], script_path))
+
+    # don't just overwrite existing scripts
+    existing_scripts = [s for (_, s) in ecs_and_script_paths if os.path.exists(s)]
+    if existing_scripts:
+        if build_option('force'):
+            _log.info("Found existing scripts, overwriting them: %s", ' '.join(existing_scripts))
+        else:
+            raise EasyBuildError("One or more scripts already exists, not overwriting them (unless forced): %s",
+                                 ' '.join(existing_scripts))
+
+    orig_env = copy.deepcopy(os.environ)
+
+    for ec, script_path in ecs_and_script_paths:
+        # obtain EasyBlock instance
+        app_class = get_easyblock_class(ec['easyblock'], name=ec['name'])
+        app = app_class(ec)
+
+        # mimic dry run, and keep quiet
+        app.dry_run = app.silent = app.toolchain.dry_run = True
+
+        # prepare build environment (in dry run mode)
+        app.check_readiness_step()
+        app.prepare_step(start_dir=False)
+
+        # compose script
+        ecfile = os.path.basename(ec.path)
+        script_lines = [
+            "#!/bin/bash",
```

Comment:
it's not, but I included it anyway as a kind of self-documentation (to make clear it's a bash script)